tests/rnd/00-test-battery-chunking.t

```diff
@@ -16,27 +16,26 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #-------------------------------------------------------------------------------
 # Test the run-functional-tests.sh --chunk option
-. $(dirname $0)/test_header
-NO_CHUNKS=2
-set_test_number $(( 2 + $NO_CHUNKS ))
+. "$(dirname "$0")/test_header"
+N_CHUNKS=2
+set_test_number "$(( 2 + N_CHUNKS ))"
 #-------------------------------------------------------------------------------
 # list all tests
 DRY_TEST_NAME="$TEST_NAME_BASE-all"
-run_ok "$DRY_TEST_NAME" $CYLC_REPO_DIR/etc/bin/run-functional-tests.sh --dry
+CTB="${CYLC_REPO_DIR}/etc/bin/run-functional-tests.sh"
+run_ok "${DRY_TEST_NAME}" "${CTB}" --dry './tests'
 # list tests for each chunk (from prove not run-functional-tests.sh)
-temp_file=$(mktemp)
-for chunk_no in $(seq $NO_CHUNKS); do
-    TEST_NAME="$TEST_NAME_BASE-chunk-$chunk_no"
-    run_ok "$TEST_NAME" \
-        $CYLC_REPO_DIR/etc/bin/run-functional-tests.sh \
-            --chunk "$chunk_no/$NO_CHUNKS" --dry
-    cat "$TEST_NAME.stdout" >> "$temp_file"
+for i_chunk in $(seq "${N_CHUNKS}"); do
+    TEST_NAME="${TEST_NAME_BASE}-chunk-${i_chunk}"
+    run_ok "${TEST_NAME}" env CHUNK="${i_chunk}/${N_CHUNKS}" "${CTB}" --dry
+    cat "${TEST_NAME}.stdout" >>'chunks.out'
```

Comment:
Test battery already runs with a PWD in a temporary location, so there is no need to use `mktemp` for output files. So `chunks.out` will be cleaned up automatically.